Solutions/GitLab/Analytic Rules/GitLab_Impersonation.yaml

```diff
@@ -0,0 +1,33 @@
+id: 0f4f16a2-b464-4c10-9a42-993da3e15a40
+name: GitLab - User Impersonation
+description: |
+  'This queries GitLab Audit Logs for user impersonation. A malicious operator or a compromised admin account could leverage the impersonation feature of GitLab to change code or
+  repository settings bypassing usual processes. This hunting queries allows you to track the audit actions done under impersonation.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes: 
+      - Syslog
+queryFrequency: 1h
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Persistence
+relevantTechniques:
+  - T1078
+query: |
+  let impersonationStart = (GitLabAudit
+  | where CustomMessage == 'Started Impersonation'
+  | extend AuthorID = tostring(AuthorID), TargetID = tostring(AuthorID));
```

Comment:
Seems like the parser should be casting to string, then this doesn't need to be done in very query.